autonomy_core/map_plan/mapper/src/local_global_mapper.cpp

```diff
@@ -288,9 +288,38 @@ void LocalGlobalMapperNode::processCloud(
                                             T_odom_lidar.translation().y(),
                                             T_odom_lidar.translation().z());
 
+  const Eigen::Vector3d global_center(global_map_cx_,
+                                      global_map_cy_,
+                                      global_map_cz_);
+
   ros::Time t0;
   cpu_times tc;
 
+  // Reset global and storage maps if the robot is far from the map centers
+  const auto diff = lidar_position_odom - global_center;
+  const auto distance = diff.norm();
+  const auto threshold = std::min(global_map_dim_d_x_, global_map_dim_d_x_) / 40;
```

Comment:
Dividing by 40 will make the update small. Maybe we should make this threshold a parameter in the mapper YAML file, maybe name it something like map_reinit_distance, and make the default value -1 (which means no reinitialization at all).